torchmetrics/functional/classification/cosine_similarity.py

```diff
@@ -0,0 +1,63 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple
+
+import torch
+from torch import Tensor
+
+
+def _cosine_similarity_update(
+    preds: Tensor,
+    target: Tensor,
+) -> Tuple[Tensor, int]:
+    preds.float()
+    target.float()
+
```

Comment:
call this function here to check that they are same shape
https://github.com/PyTorchLightning/metrics/blob/21ce4f64c9446fe5b98d96ea1dab65caa81441ea/torchmetrics/utilities/checks.py#L23-L26